site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java

```diff
@@ -192,15 +204,46 @@ private HstQuery buildQuery(HstRequest request) {
             default:
                 log.error("Unknown sort mode: " + sortParam);
                 break;
-        }
+        }    
 
         return constructQuery(queryBuilder, searchStringConstraint);
     }
 
+    private HippoBean[] deriveScope(HstRequest request) {
+        SearchArea areaParam = getAreaOption(request);
+
+        final HippoBean base = RequestContextProvider.get().getSiteContentBaseBean();
+        final HippoBean newsFolder = base.getBean(FOLDER_NEWS);
+        final HippoBean publicationsFolder = base.getBean(FOLDER_PUBLICATIONS);
+        final HippoBean servicesFolder = base.getBean(FOLDER_SERVICES);
+        final HippoBean dataAndInformationFolder = base.getBean(FOLDER_DATAANDINFORMATION);
+        final HippoBean nationalIndicatorLibraryFolder = base.getBean(FOLDER_NIL);
+
+        List<HippoBean> scopeBeans = new ArrayList<HippoBean>();
+
+        if (areaParam.equals(SearchArea.NEWS)) {
```

Comment:
If you're using ENUMs you might want to use a switch here instead, more explicit.

Reply:
Agreed. Granted, that's replacing one kind of evil with another but the switch/case will be a bit easier to read.

Reply:
I'm changing to use a switch ;)